src/traffic_server/traffic_server.cc

```diff
@@ -2215,6 +2215,21 @@ main(int /* argc ATS_UNUSED */, const char **argv)
       REC_ReadConfigInteger(num_of_udp_threads, "proxy.config.udp.threads");
     }
 
+    ats_scoped_str server_ports(255);
+    *server_ports = '\0';
+    REC_ReadConfigString(server_ports, "proxy.config.http.server_ports", 255);
+    bool quic_server_port = strstr(server_ports, "/quic/") != server_ports;
+
+#if TS_USE_QUIC == 0
+    if (quic_server_port) {
+      Fatal("proxy.config.http.server_ports listening for quic but traffic_server wasn't built with quic support.");
+    }
+#endif
+
+    if (quic_server_port && num_of_udp_threads == 0) {
+      Fatal("proxy.config.http.server_ports listening for quic but proxy.config.udp.threads is 0.");
```

Comment:
I'm not sure if we should use `Fatal`. Even port configuration error is just a warning. It depends on the use case for sure, but QUIC being unavailable is not the end of the world.

Reply:
@maskit I think you might be right w.r.t warning - I bounced between fatal and warning a few times before proposing.

> but QUIC being unavailable is not the end of the world.

Well, QUIC being unavailable when you've specifically asked ATS to serve quic could be considered to be the end of the world - it would certainly be surprising enough that I'd think we'd want to catch it sooner rather than later

Reply:
How you catch misconfiguration and how ATS behave while configuration may be wrong are different things. You could watch for a warning to catch misconfiguration, or you could also watch for the UDP ports. Ideally you should periodically send health check requests. Look at the code I mentioned above. You can try to enable mptcp where mptcp is unavailable on the host and that does not stop ATS. Lack of QUIC does not threaten the safety. QUIC can be unavailable even with proper configuration if middle boxes block it. I still think Fatal is too much.